divert/core.py

```diff
@@ -16,6 +18,19 @@ def raise_again(self, *_: Any) -> bool:
         self._jumps_remaining -= 1
         return self._jumps_remaining > 0
 
+    @classmethod
+    def make_weak(cls, *args: Any, **kwargs: Any) -> Diversion:
+        """Make weak variant of the diversion class.
+
+        By dynamically forming it provides support for generic extensions of the base diversions.
```

Comment:
ok hear me out instead of doing this you have a metaclass that generates this exactly once. It's so much machinery for about no payoff

Reply:
while we're on the subject of dumb ways to implement this, how about the 3-arg form of type?
```
Weakened = type('Weakened', (Diversion, Exception,), {})
```